### PR TITLE
CI-42: Emotion Entry

### PR DESCRIPTION
**Pull request by Maryna-Harasko**
[CI-42](https://cloverpop-internship.atlassian.net/browse/CI-42)

[CI-42]: https://cloverpop-internship.atlassian.net/browse/CI-42?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
 sections, the application should allow users to enter a word of their own choosing to describe their week. The application should allow users to then categorize their created word based upon how they feel about the word/emotion. 
The application should allow administrators to ‘Approve’ any emotion pending submitted by users to the main Emotion database, under its corresponding category. When this happens, the new emotion word gets added to the list of available ones that can show up as clickable options in the Emotion Selection Web

### CODE REVIEW

File: app/controllers/api/v1/emotions_controller.rb [hunk 1]
```diff
@@ -39,8 +60,12 @@ def response_hash
   end
 
   def build_three_set
-    Emotion.positive.sample(NUMBER_OF_ELEMENTS) +
-      Emotion.neutral.sample(NUMBER_OF_ELEMENTS) +
-      Emotion.negative.sample(NUMBER_OF_ELEMENTS)
+    Emotion.where(public: true).positive.sample(NUMBER_OF_ELEMENTS) +
+      Emotion.where(public: true).neutral.sample(NUMBER_OF_ELEMENTS) +
+      Emotion.where(public: true).negative.sample(NUMBER_OF_ELEMENTS)
+  end
```
Comment (SerhiiBorozenets) [inline app/controllers/api/v1/emotions_controller.rb:66]: @Maryna-Harasko please define in emotion model scope:
`scope :public, -> { where(public: true) } `
and use it here
```
 Emotion.public.positive.sample(NUMBER_OF_ELEMENTS)
 ...
```

**Review by SerhiiBorozenets — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/emotions_controller.rb [hunk 2]
```diff
@@ -13,6 +13,27 @@ def index
     end
   end
 
+  def create
+    @emotion = Emotion.new(emotion_params)
+    @emotion_existed = Emotion.all.find_by(word: params.dig('emotion', 'word'))
```
Comment (SBorozenets) [inline app/controllers/api/v1/emotions_controller.rb:18]: @Maryna-Harasko you shouldn`t use instance variable (with @) if it used only in one method.

File: app/javascript/components/Pages/EmotionEntry.js [hunk 3]
```diff
@@ -1,26 +1,136 @@
...
+    if (category === "positive") {
+      return "input-positive";
+    } else if (category === "negative") {
+      return "input-negative";
+    } else if (category === "neutral") {
+      return "input-neutral";
+    } else {
+      return "";
+    }
+  };
```
Comment (SBorozenets) [inline app/javascript/components/Pages/EmotionEntry.js:65]: @Maryna-Harasko for big conditions in js more convenient use switch
https://medium.com/nerd-for-tech/a-case-to-switch-using-switch-statements-in-react-e83e01154f60

**Review by SBorozenets — CHANGES_REQUESTED** [review]

**Review by SerhiiBorozenets — APPROVED** [review]

**Review by SBorozenets — APPROVED** [review]

File: app/assets/stylesheets/emition_entry.scss [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  margin-bottom: 32px !important;
+  color: $black;
+}
+
+.emotion-entry__h4{
+  font-weight: 400;
+  color: $gray-600;
+}
+
+.input__new-word{
```
Comment (SokalMet) [inline app/assets/stylesheets/emition_entry.scss:14]: I think this class name should look like input-new-word

File: app/assets/stylesheets/emition_entry.scss [hunk 5]
```diff
@@ -0,0 +1,133 @@
+@import "colors";
+@import "design_system_vibereport";
+
+.emotion-entry__h1{
+  margin-bottom: 32px !important;
+  color: $black;
+}
+
+.emotion-entry__h4{
```
Comment (SokalMet) [inline app/assets/stylesheets/emition_entry.scss:9]: no point to make difficult class names here. check prev comment

File: app/assets/stylesheets/emition_entry.scss [hunk 6]
```diff
@@ -0,0 +1,133 @@
+@import "colors";
+@import "design_system_vibereport";
+
+.emotion-entry__h1{
```
Comment (SokalMet) [inline app/assets/stylesheets/emition_entry.scss:4]: I think this class name should look like emotion-entry with parent tag h1:
h1 {
.emotion-entry {.....}
}

no point to make difficult class names here

**Review by SBorozenets — APPROVED** [review]

**Review by SokalMet — APPROVED** [review]

**Pull request merged**